aten/src/ATen/native/cpu/ScatterGatherKernel.cpp

```diff
@@ -292,9 +274,9 @@ struct cpu_scatter_gather_base_kernel {
   }
 };
 
-void gather_cpu_kernel(Tensor& result, const Tensor& self, int64_t dim, const Tensor& index) {
+void gather_cpu_kernel(const Tensor& result, const Tensor& self, int64_t dim, const Tensor& index) {
   cpu_scatter_gather_base_kernel</*is_scatter_like=*/false>()(
-    result, dim, index, self,
+    const_cast<Tensor&>(result), dim, index, self,
```

Comment:
I think it's better to propagate this `const` instead of casting it here.

Reply:
Done, thanks for pointing this out.